### PR TITLE
Add a changelog.

### PR DESCRIPTION
**Pull request by beccatortell**
Important change #1: Changes required for SignalFlow support.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+Each client's features are now divided among data transmission to SignalFx (`ingest`), metadata retrieval (`rest`), and data streaming from SignalFx to your client (`signalflow`).  This means that when you instantiate a SignalFx client object, you must also choose which SignalFx API you will access using that object.
+
+- If you use the client to send data to SignalFx, use `ingest`. 
+- If you use the client to retrieve metric names and metadata from SignalFx, use `rest`. 
+- To use the new SignalFlow API to stream analytics to your client, use `signalflow`. 
+
+Before SignalFlow support, client instantiation used to look like this:
+
+```
```
Comment (mpetazzoni) [inline CHANGELOG.md]: Use
```
```python
```

(Ditto for all other Python snippets)

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
+### CHANGELOG
+
+This file documents important changes to the SignalFx Python client library. 
+
+#### [0.4.0] - 2016-04-14: Support for SignalFlow API
+
+In version 0.4.0 of this client, we introduced support for the SignalFlow API. This means you can use this client to programmatically stream analytical computations from SignalFx in real time, in addition to sending data in to SignalFx. Using SignalFlow, you can build your own applications that leverage SignalFx's streaming analytics outside the SignalFx UI. To read more about SignalFlow, click here:  https://developers.signalfx.com/docs/signalflow-overview
```
Comment (mpetazzoni) [inline CHANGELOG.md]: Wrap long lines to 80 chars

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
+### CHANGELOG
```
Comment (mpetazzoni) [inline CHANGELOG.md]: Any particular reason why you're starting at h3 ?

File: CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+This file documents important changes to the SignalFx Python client library. 
+
+#### [0.4.0] - 2016-04-14: Support for SignalFlow API
```
Comment (mpetazzoni) [inline CHANGELOG.md]: We never released a 0.4.0 version. Date is also wrong

File: CHANGELOG.md (hunk 3 above)
Comment (beccatortell) [inline CHANGELOG.md]: Only that it looks less offensively huge when rendered.

File: CHANGELOG.md (hunk 4 above)
Comment (beccatortell) [inline CHANGELOG.md]: Can you please provide correct values?

**Comment by mpetazzoni** [comment]
Thanks!

**Pull request merged**